wayland/install-wayland.txt

```diff
@@ -0,0 +1,158 @@
+
+That is entirely up to you. Aside from just trying out something new, a Wayland
+based GUI _can_ simplify your system. Screen tearing is not possible by design
+and hardware accelerated rendering may happen in places which are not covered by
+Xorg.
+
+- less moving parts [13], although many graphical programs still heavily rely on
+  Xorg libraries. (Mesa [16], Firefox [17], Qt-WebEngine [18], ...) 
+- No screen tearing.
+- Hardware accelerated video playback in Firefox. (VAAPI)
+- Hardware accelerated rendering in webkit2gtk. (WPEBackend-fdo)
+- If you can delimit your needs and expectations on KISS in the aforementioned 
+  caveats.
+
+
+Prerequisites
+________________________________________________________________________________
+
+Though KISS does not support Wayland officially there is no problem in making it
+capable through personal or 3rd party repositories. [19] The official
+repositories satisfy all dependencies. After "wayland" and "wayland-protocols"
+are installed, some package adjustments are required to make them to known
+citizens of your system.
+
+NOTE: From this point it is expected that "wayland" and "wayland-protocols" are
+      installed and you have forked the required packages with modified
+      configure flags at the beginning of KISS_PATH. [20] Managing repositories
+
+* Packages which need to be forked [21]:
+     
+      firefox --enable-default-toolkit=cairo-gtk3-wayland
+      gtk+3   --enable-wayland-backend
+      mesa    -Dplatforms=x11,drm,wayland
+
+
+* Packages which pick up Wayland automatically at compile time [22]:
+  [needs expansion and verification]
+
+      cairo 
+      intel-vaapi-driver 
+      libva 
+      mpv 
+      pango 
+      sdl2 
+      webkit2gtk
+
+
+Install Wayland on top Xorg KISS
+________________________________________________________________________________
+
+Mesa must be rebuilt first, followed by the aforementioned packages [21] [22] in
+a dependency given ascending order.
+
+NOTE: kiss-revdepends is the tool you need.
+
+mesa, cairo, pango, gtk+3, firefox/webkit2gtk
+      qt5-wayland
+      mpv
+      libva, intel-vaapi-driver
+      sdl2
+  
+$ kiss b mesa && kiss i mesa && kiss b pango && kiss i pango ...
+
+
+Install Wayland on top base KISS
+________________________________________________________________________________
+
+Starting off a fresh base installation there is no manual intervention needed as
+everything _should_ be autoconfigured.
+[needs reproduction]
+
+
+Post installation
+________________________________________________________________________________
+
+* In environments without systemd/elogind/consolekit the XDG_RUNTIME_DIR
+  variable must be set manually. [23] Export this in your shellrc.
+
+  $ export XDG_RUNTIME_DIR=${XDG_RUNTIME_DIR:-/tmp/$(id -u)-runtime-dir}
+  $ 
+  $ [ -d "$XDG_RUNTIME_DIR" ] || {
+  $     mkdir -p   "$XDG_RUNTIME_DIR"
+  $     chmod 0700 "$XDG_RUNTIME_DIR"
+  $ }
```

Comment:
This might be better without the `$` at the start of each line. Will make copy/paste easier.